app/policies/ebook_operation.rb

```diff
@@ -20,8 +20,24 @@ def unrestricted?
     end
 
     def licensed_for?(entitlement)
-      authority
-        .licenses_for(actor, ebook)
-        .any? { |license| license.allows?(entitlement) }
+      licenses = authority.licenses_for(actor, ebook)
+
+      return true if licenses
+                       .where(licensee_type: "Greensub::Individual")
+                       .any? { |license| license.allows?(entitlement) } && actor.individual.present?
+
+      return true if licenses
+                       .where(licensee_type: "Greensub::Institution")
+                       .any? { |license| license.allows?(entitlement) } && affiliation_match?(licenses)
```

Comment:
Need to match the actor's institution affiliations per each license.

.any? { |license| license.allows?(entitlement) && license.affiliations_match?(actor.affiliations) }